# Lecture 7 : intro to Pandas

# Report card dictionary

In [1]:
report_card_dictionary = { "student_names": ["Niramay","aryan", "yash", "shubham"],
                          "math_scores": [45,42,41,40],
                          "bio_scores": [50,45,30,80],
                          "history_score": [80,100,90,85],
                          "physics_scores": [80,58,90,45]
                         }
report_card_dictionary
                          
                   

{'student_names': ['Niramay', 'aryan', 'yash', 'shubham'],
 'math_scores': [45, 42, 41, 40],
 'bio_scores': [50, 45, 30, 80],
 'history_score': [80, 100, 90, 85],
 'physics_scores': [80, 58, 90, 45]}

# set up

In [2]:
import pandas as pd

# covert dictionary to data frame

In [3]:
report_card_df = pd.DataFrame(report_card_dictionary)
report_card_df

#report_card_df.head() if data is too long

student_names  math_scores  bio_scores  history_score  physics_scores
0       Niramay           45          50             80              80
1         aryan           42          45            100              58
2          yash           41          30             90              90
3       shubham           40          80             85              45

In [4]:
report_card_df.head()

student_names  math_scores  bio_scores  history_score  physics_scores
0       Niramay           45          50             80              80
1         aryan           42          45            100              58
2          yash           41          30             90              90
3       shubham           40          80             85              45

In [5]:
report_card_df.head(2)

student_names  math_scores  bio_scores  history_score  physics_scores
0       Niramay           45          50             80              80
1         aryan           42          45            100              58

In [6]:
# Sample of 2 rows in data frame it will give random rows

report_card_df.sample(2)

student_names  math_scores  bio_scores  history_score  physics_scores
0       Niramay           45          50             80              80
1         aryan           42          45            100              58

# Create a discription of our numeric columns

In [7]:
report_card_df.describe()

math_scores  bio_scores  history_score  physics_scores
count     4.000000    4.000000       4.000000        4.000000
mean     42.000000   51.250000      88.750000       68.250000
std       2.160247   20.966243       8.539126       20.467454
min      40.000000   30.000000      80.000000       45.000000
25%      40.750000   41.250000      83.750000       54.750000
50%      41.500000   47.500000      87.500000       69.000000
75%      42.750000   57.500000      92.500000       82.500000
max      45.000000   80.000000     100.000000       90.000000

# Add Colums called total score

In [8]:
report_card_df["total_score"] = report_card_df["math_scores"] + report_card_df["bio_scores"] + report_card_df["history_score"] + report_card_df["physics_scores"]
report_card_df

student_names  math_scores  bio_scores  history_score  physics_scores  \
0       Niramay           45          50             80              80   
1         aryan           42          45            100              58   
2          yash           41          30             90              90   
3       shubham           40          80             85              45   

   total_score  
0          255  
1          245  
2          251  
3          250

# Create another colum called avg score

In [9]:
report_card_df["average_score"] = report_card_df["total_score"]/4
report_card_df

student_names  math_scores  bio_scores  history_score  physics_scores  \
0       Niramay           45          50             80              80   
1         aryan           42          45            100              58   
2          yash           41          30             90              90   
3       shubham           40          80             85              45   

   total_score  average_score  
0          255          63.75  
1          245          61.25  
2          251          62.75  
3          250          62.50

# Describe the report card df

In [10]:
report_card_df.describe()

math_scores  bio_scores  history_score  physics_scores  total_score  \
count     4.000000    4.000000       4.000000        4.000000     4.000000   
mean     42.000000   51.250000      88.750000       68.250000   250.250000   
std       2.160247   20.966243       8.539126       20.467454     4.112988   
min      40.000000   30.000000      80.000000       45.000000   245.000000   
25%      40.750000   41.250000      83.750000       54.750000   248.750000   
50%      41.500000   47.500000      87.500000       69.000000   250.500000   
75%      42.750000   57.500000      92.500000       82.500000   252.000000   
max      45.000000   80.000000     100.000000       90.000000   255.000000   

       average_score  
count       4.000000  
mean       62.562500  
std         1.028247  
min        61.250000  
25%        62.187500  
50%        62.625000  
75%        63.000000  
max        63.750000

#  Insights 
- what is mean math score?42
- what is lowest bio score?30
- what is highest history score?100
- what is mean total score?250.25
- what is mean average score?62.562

# Report card insights

In [11]:
report_card_insights_df = report_card_df.describe()
report_card_insights_df


math_scores  bio_scores  history_score  physics_scores  total_score  \
count     4.000000    4.000000       4.000000        4.000000     4.000000   
mean     42.000000   51.250000      88.750000       68.250000   250.250000   
std       2.160247   20.966243       8.539126       20.467454     4.112988   
min      40.000000   30.000000      80.000000       45.000000   245.000000   
25%      40.750000   41.250000      83.750000       54.750000   248.750000   
50%      41.500000   47.500000      87.500000       69.000000   250.500000   
75%      42.750000   57.500000      92.500000       82.500000   252.000000   
max      45.000000   80.000000     100.000000       90.000000   255.000000   

       average_score  
count       4.000000  
mean       62.562500  
std         1.028247  
min        61.250000  
25%        62.187500  
50%        62.625000  
75%        63.000000  
max        63.750000

In [12]:
report_card_insights_df = report_card_df.describe().transpose()
report_card_insights_df


count      mean        std     min       25%      50%     75%  \
math_scores       4.0   42.0000   2.160247   40.00   40.7500   41.500   42.75   
bio_scores        4.0   51.2500  20.966243   30.00   41.2500   47.500   57.50   
history_score     4.0   88.7500   8.539126   80.00   83.7500   87.500   92.50   
physics_scores    4.0   68.2500  20.467454   45.00   54.7500   69.000   82.50   
total_score       4.0  250.2500   4.112988  245.00  248.7500  250.500  252.00   
average_score     4.0   62.5625   1.028247   61.25   62.1875   62.625   63.00   

                   max  
math_scores      45.00  
bio_scores       80.00  
history_score   100.00  
physics_scores   90.00  
total_score     255.00  
average_score    63.75

# Creating a csv file of our report card insight means to covert the ablove list into excel

In [13]:
report_card_insights_df.to_csv("report card insights.csv")